lib/src/utils/current_locale.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:intl/intl.dart';
+import 'package:lichess_mobile/src/model/settings/general_preferences.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'current_locale.g.dart';
+
+@riverpod
+String currentLocale(CurrentLocaleRef ref) {
```

Comment:
I think this is not needed and we should not handle it this way. I presume you made it to pass the locale to the date formatters. But these are designed to use the default locale, and we just need to correctly set the default locale elsewhere.

I know currently this is broken in `main` branch, but I'm taking care of it in the challenges PR.

Anyway I'd prefer you remove that logic from this PR so we can fix it globally in another PR.

Reply:
Since the users can now change the language in settings shouldn't we format dates accordingly ?

Reply:
Yes but as explained it should work without passing an argument to the formatter. And we should not deal with that in this PR.

Reply:
Yes I'll remove it from this PR but just to let you know, if I remember correctly I chose to do it like this because if you just change the default locale it doesn't apply to already loaded pages. I remember that I tried both solutions and I ended it with this one but I understand that it doesn't add much for the problems it might cause.

Reply:
The solution, I think, is to set again the `Intl.defaultLocale`, I'm going to do it in the challenge PR.

Reply:
That's what I meant when I said > change the default locale